index.html

```diff
@@ -4292,6 +4306,493 @@ <h3>Proof Formats</h3>
+related to the verification failure.
+              </li>
+            </ol>
+          </li>
+          <li>
+Return <var>verified</var>.
+          </li>
+        </ol>
+
+        <p>
+The steps for verifying the state of the securing mechanism and verifying
+that the input document is a <a>conforming document</a> MAY be performed in
+a different order than the one provided above as long as the
+implementation returns errors for the same inputs.
+Implementations MAY produce different errors than described above.
+        </p>
+
+      </section>
+
+      <section class="normative">
+        <h3>Verify Data Model</h3>
+
+        <p>
+The algorithm in this section can be used to determine if a provided document is
+conformant to the data model provided in this specification. This algorithm
+takes a document (object <var>inputDocument</var>) and produces an object
+(<var>dataModelVerified</var>) that contains the following:
+        </p>
+
+        <ul>
+          <li>
+a verification status (boolean <var>status</var>)
+          </li>
+          <li>
+a verified document (object <var>document</var>)
+          </li>
+          <li>
+zero or more warnings (array of <a>ProblemDetails</a> <var>warnings</var>)
+          </li>
+          <li>
+zero or more errors (array of <a>ProblemDetails</a> <var>errors</var>)
+          </li>
+        </ul>
+
+        <p>
+The data model verification algorithm is as follows:
+        </p>
+
+        <ol class="algorithm">
+          <li>
+Set <var>dataModelVerified</var> to an empty object. Initialize the following
+fields in <var>dataModelVerified</var>: <var>status</var> to `false`,
+<var>warnings</var> to an empty array, and <var>errors</var> to an empty array.
+          </li>
+          <li>
+Ensure that the `@context` property of <var>document</var> conforms to the
+statements in <a href="#contexts"></a>. If not, return
+<var>dataModelVerified</var>, which will contain a false verification status.
+          </li>
+          <li>
+Starting at the top level, recursively process each object <var>obj</var> in
+<var>document</var>, setting <a href="#MALFORMED_VALUE_ERROR">
+MALFORMED_VALUE_ERROR</a> errors in
+<var>dataModelVerified</var>.<var>errors</var> if errors are found, by
+performing the following checks:
+            <ol class="algorithm">
+              <li>
+If present, ensure that the `id` property conforms to the "MUST" statements in
+Section <a href="#identifiers"></a>.
+              </li>
+              <li>
+If present, ensure that the `type` property conforms to the "MUST" statements in
+Section <a href="#types"></a>.
+              </li>
+              <li>
+If present, ensure that the `name` and/or `description` properties conform to
+the "MUST" statements in Section <a href="#names-and-descriptions"></a>.
+              </li>
+              <li>
+If present, ensure that the `proof` property conforms to
+the "MUST" statements in Section <a href="#securing-mechanisms"></a>.
```

Comment:
This will produce a formal objection from Transmute if it remains in the core data model.

We can't have every securing mechanism forced to check the `proof` property.

For any verifier who does not intend to support data integrity proofs, `proof` is just another undefined / issuer dependent term, that is ignored when its not understood.

I think its probably possible to address this without all the text that is written here, but it might require moving some of this text to data integrity and calling out to data integrity for a boolean response to the question "verified".

We cannot mix `proof` validation with verification like this.

Reply:
Removed in ebf2d23bf1d6e290d1e6087da67fc41965fa70d6, but for different reasons (it's checked by the subspecs).

To be clear, this was here to check document conformance, for the statements that exist in this specification, for the `proof` field and structure (which requires `type` to exist). That will also be checked by the Data Integrity specifications when they come into play.

Reply:
Removed in ebf2d23bf1d6e290d1e6087da67fc41965fa70d6.

Reply:
Thanks! Resolving.